### ISSUE TITLE
Extend SecHub CodeScan report parts to have an explanation field  inside JSON report

### ISSUE THREAD

**Issue by de-jcup** [Daimler/sechub#474]
> As a developer having a SecHub JSON report, I want to have an explanation field in JSON about the finding.

**TODOs**
- change the internal sereco report model
- provide explanation field(s) also in sechub report JSON model

**Still to DECIDE**
- decide which kind of format we will put in (e.g. many products provide a HTML explanation - do we really want to insert HTML here?!?!)
- decide structure e.g. just a
```json
 "explanation" : "i am the explanation"  
```
or something like
```json
"explanation": {
         "text": "plain text" , 
         "html" : "html content", 
         "links": ["https://owasp.org/topten/xyz"]
 } 
```
the latter one would be nice because we could provide different possibilities (of course those should be optional).
But it would be easier to implement and use - e.g. in plugins - so prefereable

---
<sup>Albert Tregnaghi <albert.tregnaghi@daimler.com>, Daimler TSS GmbH, [imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)</sup>

**Comment by Jeeppler** [comment]
In my opinion, we should not support HTML itself. However, besides plain text we could support massages in AsciiDoc and/or markdown. For an inspiration: the SARIF format supports messages in plain text and markdown: https://github.com/microsoft/sarif-tutorials/blob/main/docs/3-Beyond-basics.md#more-about-messages

In addition, it would be nice to have an optional language field, for both the explanation text (description) and the links. Maybe, the explanation can look like this:

~~~
"explanation": {
         "descriptions": [
                {
                  "text": "plain text" , 
                  "asciidoc" : "asciidoc content",
                  "markdown": "markdown content",
                  "language": "de"
                },
                {
                  "text": "by default we assume English as language for a message"
                }
          ],
         "links": [
                 {
                   "link": "https://owasp.org/de/topten/xyz",
                   "language": "de",
                },
                {
                   "link": "https://owasp.org/topten/xyz"
                }
          ]
 }
~~~

**Comment by de-jcup** [comment]
Will be solved with #510
